cibuildwheel/pyodide.py

```diff
@@ -0,0 +1,385 @@
+from __future__ import annotations
+
+import contextlib
+import os
+import shutil
+import sys
+from collections.abc import Sequence, Set
+from dataclasses import dataclass
+from pathlib import Path
+
+from filelock import FileLock
+
+from .architecture import Architecture
+from .environment import ParsedEnvironment
+from .logger import log
+from .options import Options
+from .typing import PathOrStr
+from .util import (
+    CIBW_CACHE_PATH,
+    AlreadyBuiltWheelError,
+    BuildFrontendConfig,
+    BuildSelector,
+    NonPlatformWheelError,
+    call,
+    download,
+    extract_zip,
+    find_compatible_wheel,
+    get_pip_version,
+    prepare_command,
+    read_python_configs,
+    shell,
+    split_config_settings,
+    test_fail_cwd_file,
+    virtualenv,
+)
+
+
+@dataclass(frozen=True)
+class PythonConfiguration:
+    version: str
+    identifier: str
+    pyodide_version: str
+    emscripten_version: str
+
+
+def install_emscripten(tmp: Path, version: str) -> Path:
+    url = "https://github.com/emscripten-core/emsdk/archive/main.zip"
```

Comment:
We could pull the version from the same toml file. Will wait for @hoodmane to comment, I’d guess it was left over from development?

Reply:
Sure it is fine to pull from the toml file. emsdk is just an installer for emscripten and its dependencies, most commits add metadata to install a new version of emscripten. I don't think installing from the main branch would ever be wrong because they never commit anything that breaks installing files. But if people prefer to select the tag corresponding to our emscripten version that should work fine too...